java/code/src/com/redhat/rhn/domain/kickstart/KickstartData.java

```diff
@@ -1573,20 +1576,30 @@ public RegistrationType getRegistrationType(User user) {
             return RegistrationType.getDefault();
         }
 
-        return RegistrationType.find((String)prof.getKsMeta().get(
+        Optional<Map<String, Object>> ksMeta = prof.getKsMeta();
+        if (ksMeta.isEmpty()) {
+            return RegistrationType.getDefault();
+        }
+        return RegistrationType.find((String)ksMeta.get().get(
```

Comment:
nitpick but this can be reduced to 
```java
       return prof.getKsMeta()
                .map(ksm -> RegistrationType.find((String)ksm.get(RegistrationType.COBBLER_VAR)))
                .orElseGet(RegistrationType::getDefault)
```
getting rid of the extra variable and .get() call.